### PR TITLE
Update LAPACK-3.9.1-GCC-10.2.0.eb

### PR DESCRIPTION
**Pull request by surak**
The original URL was failing

### CODE REVIEW

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso
EB_ARGS="--force-download --sourcepath /tmp/$USER/pr15047 --installpath /tmp/$USER/pr15047"

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
jsczen2g1.int.jsc-zen2.easybuild-test.cluster - Linux Rocky Linux 8.5, x86_64, AMD EPYC 7742 64-Core Processor (zen2), 1 x NVIDIA GRID V100-4C, 460.73.01, Python 3.6.8
See https://gist.github.com/b8a820fef88edb9c1f7c62d06b26198f for a full test report.

**Review by SebastianAchilles — APPROVED** [review]
lgtm

File: easybuild/easyconfigs/l/LAPACK/LAPACK-3.9.1-GCC-10.2.0.eb [hunk 1]
```diff
@@ -9,7 +9,7 @@ description = """LAPACK is written in Fortran90 and provides routines for solvin
 toolchain = {'name': 'GCC', 'version': '10.2.0'}
 toolchainopts = {'pic': True}
 
-source_urls = ['https://github.com/Reference-LAPACK/lapack/archive/']
+source_urls = ['https://github.com/Reference-LAPACK/lapack/archive/refs/tags']
```
Comment (boegel) [inline easybuild/easyconfigs/l/LAPACK/LAPACK-3.9.1-GCC-10.2.0.eb:12]: This change doesn't make sense to me, the original URL is working fine?!

Comment (surak) [reply easybuild/easyconfigs/l/LAPACK/LAPACK-3.9.1-GCC-10.2.0.eb:12]: It was 404 to me on two machines.

Comment (SebastianAchilles) [reply easybuild/easyconfigs/l/LAPACK/LAPACK-3.9.1-GCC-10.2.0.eb:12]: The new URLs is used on the release page: https://github.com/Reference-LAPACK/lapack/releases
That is why I think it makes sense to update.

**Pull request merged**